packages/react-form/src/tests/useField.test-d.tsx

```diff
@@ -65,3 +65,44 @@ it('should type onChange properly', () => {
     )
   }
 })
+
+it('should type array subfields', () => {
+  type FormDefinition = {
+    nested: {
+      people: {
+        name: string
+        age: number
+      }[]
+    }
+  }
+
+  function App() {
+    const form = useForm({
+      defaultValues: {
+        nested: {
+          people: [],
+        },
+      } as FormDefinition,
```

Comment:
Just a suggestion, and maybe not a very important one, but we could use `satisfied`  if we added an element to the `people` array in `defaultValues`.  This way we'd get rid of the type casting.

```suggestion
      defaultValues: {
        nested: {
          people: [{ name: 'Tom', age: 50 }],
        },
      } satisfies FormDefinition,
```